c10/core/impl/cow/state.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <c10/core/impl/cow/simulator.h>
+#include <c10/macros/Macros.h>
+#include <c10/util/Optional.h>
+#include <c10/util/intrusive_ptr.h>
+
+#include <cstdint>
+#include <mutex>
+
+namespace c10::impl::cow {
+
+// Responsible for managing the copy-on-write simulation state
+// machine.
+class C10_API State {
```

Comment:
nit: prefer a more descriptive name here

Reply:
updated to cow::StateMachine.